consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -614,45 +791,233 @@ impl PruningProofManager {
+
+                while !self.reachability_service.is_dag_ancestor_of(common_ancestor.hash, block_at_depth_2m) {
+                    common_ancestor = match self.find_selected_parent_header_at_level(&common_ancestor, level) {
+                        Ok(header) => header,
+                        // Try to give this last header a chance at being root
+                        Err(PruningProofManagerInternalError::NotEnoughHeadersToBuildProof(_)) => break,
+                        Err(e) => return Err(e),
+                    };
+                }
+
+                common_ancestor.hash
+            };
+
+            if level == 0 {
+                return Ok((self.ghostdag_store.clone(), selected_tip, root));
+            }
+
+            // Step 3 - Fill the ghostdag data from root to tip
+            let ghostdag_store = Arc::new(DbGhostdagStore::new_temp(temp_db.clone(), level, cache_policy, cache_policy, tries));
+            let has_required_block = self.fill_level_proof_ghostdag_data(
+                root,
+                pp_header.header.hash,
+                &ghostdag_store,
+                Some(block_at_depth_m_at_next_level),
+                level,
+            );
+
+            // Step 4 - Check if we actually have enough depth.
+            // Need to ensure this does the same 2M+1 depth that block_at_depth does
+            if has_required_block
+                && (root == self.genesis_hash || ghostdag_store.get_blue_score(selected_tip).unwrap() >= required_level_depth)
+            {
+                break Ok((ghostdag_store, selected_tip, root));
+            }
+
+            tries += 1;
+            if is_last_level_header {
+                if has_required_block {
+                    // Normally this scenario doesn't occur when syncing with nodes that already have the safety margin change in place.
+                    // However, when syncing with an older node version that doesn't have a safety margin for the proof, it's possible to
+                    // try to find 2500 depth worth of headers at a level, but the proof only contains about 2000 headers. To be able to sync
+                    // with such an older node. As long as we found the required block, we can still proceed.
+                    warn!("Failed to find sufficient root for level {level} after {tries} tries. Headers below the current depth of {required_base_level_depth} are already pruned. Required block found so trying anyway.");
+                    break Ok((ghostdag_store, selected_tip, root));
+                } else {
+                    panic!("Failed to find sufficient root for level {level} after {tries} tries. Headers below the current depth of {required_base_level_depth} are already pruned");
+                }
+            }
+
+            // If we don't have enough depth now, we need to look deeper
+            required_base_level_depth <<= 1;
```

Comment:
Maybe we should do a more subtle increase? 
```suggestion
            required_base_level_depth = (required_base_level_depth as f64 * 1.1) as u64;
```

It's still gonna be a logarithmic number of tries, although with a smaller base. Another option is to increase the multiplier (1.1) at each iteration by 0.05 and bound it by 2.0